Merge upstream

trombik opened the pull request:
update `reallyenglish-master` to the latest upstream master.

this PR is for `reallyenglish-master`.

Commit:
introduce Debian 8.8

Commit:
introduce Debian 8.8

Commit:
update archlinux iso_url

Commit:
update archlinux iso_url

Commit:
another iso_url fix

Commit:
another iso_url fix

Commit:
use mirror

Commit:
use mirror

Commit:
update iso_checksum

Commit:
update iso_checksum

Commit:
drop Debian 8.7

as requested in a comment
https://github.com/kaorimatz/packer-templates/pull/18#issuecomment-305944843

Commit:
drop Debian 8.7

as requested in a comment
https://github.com/kaorimatz/packer-templates/pull/18#issuecomment-305944843

kaorimatz committed:
Merge pull request #18 from reallyenglish/debian_8.8

introduce Debian 8.8

enderqa committed:
Update Debian 8 mirror URL

Debian 8 ISO moved to new location. Updated URL.

enderqa committed:
Update Debian 8 mirror URL

Debian 8 ISO moved to new location. Updated URL.

kaorimatz committed:
Merge pull request #20 from enderqa/fix_debian8_url

Update Debian 8 mirror URL

enderqa committed:
Debian 9.0 json template and preseed.

For building Debian 9.

kaorimatz committed:
Debian 9.0: disable predictable network interface naming

kaorimatz committed:
Debian 9.0: use open-vm-tools instead of vmware tools

kaorimatz committed:
VyOS: fix the package repository URL

kaorimatz committed:
VyOS: create vagrant group and add vagrant user to it

kaorimatz committed:
Merge remote-tracking branch 'origin/debian-9.0-amd64'

kaorimatz committed:
Arch Linux: update to 2017.07.01 ISO

Commit:
Merge branch 'master' into upstream/master

Commit:
Merge pull request #39 from reallyenglish/upstream/master

update master branch to the latest upstream

Commit:
Merge branch 'master' into merge_upstream

mitsururike commented on debian-9.0-amd64.json:
```diff
@@ -27,20 +28,20 @@
     "iso_checksum_type": "{{user `iso_checksum_type`}}",
-    "output_directory": "output-debian-8.7-amd64-{{build_type}}",
-    "vm_name": "packer-debian-8.7-amd64",
-    "hard_drive_interface": "scsi",
```
this must be merged.

mitsururike commented on http/debian-9.0/preseed.cfg:
```diff
@@ -52,10 +52,7 @@ d-i preseed/early_command string
         echo 'Defaults:vagrant !requiretty' > /target/etc/sudoers.d/vagrant;      \
         echo 'vagrant ALL=(ALL) NOPASSWD: ALL' >> /target/etc/sudoers.d/vagrant;  \
-        chmod 440 /target/etc/sudoers.d/vagrant;                                  \
-        sed -i -e 's/^GRUB_TIMEOUT=.*$/GRUB_TIMEOUT=0/' /target/etc/default/grub; \
```
this must be merged.

mitsururike reviewed (commented):
debian-8.7 is used in https://github.com/reallyenglish/ansible-role-java.

trombik commented:
we really need a proper procedure to merge upstream.

Commit:
restore debian 8.7

Commit:
bump version

Commit:
update spec files

Commit:
include our modifications to debian 9.0

and improve apt.sh by using $codename

Commit:
add all platforms that we are using

Commit:
minor bugfixes

Commit:
finish the test implementation

there are still some duplicates and the test is not very safe, but work
for me.

run a single test:

```
bundle exec rake "reallyenglish:spec['openbsd-6.0-amd64']"
```

run all tests:

```
bundle exec rake "reallyenglish:test"
```

trombik commented:
there might be missing test cases. please test by following the instruction described in c1c2c9d64ea7a5c6900717c81b8225bf51d65511

the test takes a long time, probably more than an hour.

mitsururike commented:
failed to build debian-9.0
```
    virtualbox-iso: + sudo apt-get -y -t jessie-backports install ansible
    virtualbox-iso: Reading package lists...
    virtualbox-iso: E: The value 'jessie-backports' is invalid for APT::Default-Release as such a release is not available in the sources
```

Commit:
remove another hard-coded code name

trombik commented:
fixed in a8a552a
